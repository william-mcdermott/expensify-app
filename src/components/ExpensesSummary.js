import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total'

export class ExpensesSummary extends React.Component {
  render() {
    return (
      <p>Viewing {this.props.expensesCount} {this.props.expensesCount === 1 ? 'expense' : 'expenses'} totalling {numeral(this.props.expensesTotal/100).format('$0,00.00')}</p>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    expensesCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
