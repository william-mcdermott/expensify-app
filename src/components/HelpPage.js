import React from 'react';

const HelpPage = () => (
  <div>
    this is from my help page
  </div>
);

export default HelpPage;
