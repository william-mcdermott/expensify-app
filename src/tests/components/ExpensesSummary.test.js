import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('it should properly render with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={9434} />);
  expect(wrapper).toMatchSnapshot()
})

test('it should properly render with 0 or more than one expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={9434} />);
  expect(wrapper).toMatchSnapshot()
})
