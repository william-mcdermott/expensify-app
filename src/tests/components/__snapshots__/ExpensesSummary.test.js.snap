// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should properly render with 0 or more than one expense 1`] = `
<p>
  Viewing 
  2
   
  expenses
   totalling 
  $94.34
</p>
`;

exports[`it should properly render with 1 expense 1`] = `
<p>
  Viewing 
  1
   
  expense
   totalling 
  $94.34
</p>
`;
